dev/ci/user-overlays/README.md

```diff
@@ -1,8 +1,22 @@
 # Add overlays for your pull requests in this directory
 
-An overlay is a file containing very simple logic to test whether we are currently building a specific pull request or git branch (useful so that overlays work on your own fork) and which changes some of the variables whose default can be found in [`ci-basic-overlay.sh`](/dev/ci/ci-basic-overlay.sh).
+When your pull request breaks an external development we test in our CI, you
```

Comment:
How about adding something like "Overlays provide a way to use a modified version of external code when building a specific PR.  The overlay specifies which fork and branch to use for the external code.  See dev/ci/user-overlays/."?

Also, aren't there some additional steps needed to finish up?  I'm guessing the workflow is like this:
- create a fork+branch for the external code (in the external repository) with the desired changes
- create an overlay file for your PR
- once the PR compiles, get the external change merged
- get the PR merged

It seems like there is a failure window between the last 2 steps in which all Coq builds other than the specified PR could fail if the external change doesn't build with the old version of Coq source.  Maybe those types of changes should be discouraged/banned.  But if they are allowed, it's operationally awkward.  If there was a way to check whether the external PR has been merged into the external repository, then Coq could apply the patch in the other builds appropriately.  But maybe the cost of doing so is greater than the benefit.

In #6801, it looks like I need to add '()' to calls to Printer.pr_open_subgoals because I added optional parameters.  This would fall into the not pretty case I mentioned, unless I keep the old routine unchanged and define a parallel Printer.pr_open_subgoals2.

Reply:
Indeed, when a patch is not backward-compatible there is currently a problem which you identify. I plan to fix this problem by changing the way the overlay system works (cf. #6724) but here, I'm trying to document the present (far from ideal) situation.